app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,191 @@
+    //     const otherUserId = getUserIdFromChannelName(currentUserId, msg.data.channel_name);
+    //     const dmAction = makeDirectChannelVisibleIfNecessary(state, otherUserId);
+    //     if (dmAction) {
+    //         actions.push(dmAction);
+    //     }
+    // } else if (msg.data.channel_type === General.GM_CHANNEL) {
+    //     const gmActions = await makeGroupMessageVisibleIfNecessary(state, post.channel_id);
+    //     if (gmActions) {
+    //         actions.push(...gmActions);
+    //     }
+    // }
+
+    //const viewingGlobalThreads = getViewingGlobalThreads(state);
+    // const collapsedThreadsEnabled = isCollapsedThreadsEnabled(state);
+    // actions.push(receivedNewPost(post, collapsedThreadsEnabled));
+    if (!shouldIgnorePost(post)) {
+        let markAsRead = false;
+        let markAsReadOnServer = false;
+
+        if (!myChannel?.manuallyUnread) {
+            if (
+                post.user_id === currentUserId &&
+                !isSystemMessage(post) &&
+                !isFromWebhook(post)
+            ) {
+                markAsRead = true;
+                markAsReadOnServer = false;
+            } else if ((post.channel_id === currentChannelId)) { // TODO: THREADS && !viewingGlobalThreads) {
+                // Don't mark as read if we're in global threads screen
+                // the currentChannelId still refers to previously viewed channel
+                markAsRead = true;
+                markAsReadOnServer = true;
+            }
+        }
+
+        if (markAsRead && markAsReadOnServer) {
+            markChannelAsRead(serverUrl, post.channel_id);
+        } else if (markAsRead) {
+            const viewedAt = await markChannelAsViewed(serverUrl, post.channel_id, true);
+            if (viewedAt instanceof Model) {
+                models.push(viewedAt);
+            }
+        } else {
+            const unreadAt = markChannelAsUnread(serverUrl, post.channel_id, 1, msg.data.mentions, false, true);
+            if (unreadAt instanceof Model) {
+                models.push(unreadAt);
+            }
+        }
+    }
+
+    database.operator.batchRecords(models);
+}
+
+export async function handlePostEdited(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+    const currentUserId = await queryCurrentUserId(database.database);
+    const data = JSON.parse(msg.data.post);
+    const post = {
+        ...data,
+        ownPost: data.user_id === currentUserId,
+    };
+
+    try {
+        fetchPostAuthors(serverUrl, [post]);
+    } catch {
+        // Do nothing
+    }
+
+    database.operator.handlePosts({
+        actionType: ActionType.POSTS.RECEIVED_NEW,
+        order: [post.id],
+        posts: [post],
+    });
+}
+
+export function handlePostDeleted(serverUrl: string, msg: WebSocketMessage) {
+    const data = JSON.parse(msg.data.post);
+
+    removePostById(serverUrl, data.id);
```

Comment:
here we should set the post deleteAt so the UI displays `(message deleted)` as it does currently on mobile/web

Reply:
Should we add any other "sanitation" to the post? I mean... a deleted posts should be "deleted" from the client, so I guess that at least the message should be overwritten (so I cannot use any tool to read the Mattermost database and read what was on that deleted message).

Also... when do we cleanup deleted posts? Or we just don't? What I mean is that (if I remember correctly) on webapp if you refresh the page, redux refreshes, and you no longer see there was a deleted posts there. On mobile, the "deleted post" message will be there "forever".

Reply:
You tap on the post and it gets removed so is not forever.

Also we need to build a scheduled background job to cleanup stale data and to apply data retention policies, this will probably be done by Anurag at some point.

As to sanitize the post, I do agree with it and actually think is a great idea. @jupenur thoughts on this?

Reply:
Yes, scrubbing the contents of the post is definitely a good idea. Customers regularly have concerns around data stored on clients and want the ability to delete it on demand. Ideally we should also scrub any locally cached attachments.